src/overlays/actors/ovl_Boss_02/z_boss_02.c

```diff
@@ -97,186 +151,2146 @@ const ActorInit Boss_02_InitVars = {
+            if (this->unk_0146[0] == 0) {
+                if (Rand_ZeroOne() < 0.3f) {
+                    this->unk_0144 = 5;
+                    this->unk_0146[0] = 150;
+                } else {
+                    this->unk_0144 = 2;
+                    this->unk_01B0.x = randPlusMinusPoint5Scaled(3000.0f * D_809DF5B0);
+                    this->unk_01B0.z = randPlusMinusPoint5Scaled(3000.0f * D_809DF5B0);
+                    if ((fabsf(this->unk_01B0.x) < (500.0f * D_809DF5B0)) &&
+                        (fabsf(this->unk_01B0.z) < (500.0f * D_809DF5B0))) {
+                        this->unk_01B0.x = 500.0f;
+                        this->unk_01B0.z = 500.0f;
+                    }
+                    this->unk_01B0.y = Rand_ZeroFloat(800.0f * D_809DF5B0) + (200.0f * D_809DF5B0);
+                    if (D_809E0422 != 0) {
+                        this->unk_01B0.y += 3150.0f;
+                    }
+                }
+                this->unk_0164 = 0.0f;
+            }
+            return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_02/func_809DA1D0.s")
+        case 2:
+            if (this->unk_0195 != 0) {
+                phi_f2 = 700.0f;
+            } else {
+                phi_f2 = 500.0f;
+            }
+            if (sqrtf(SQ(spCC) + SQ(spC8) + SQ(spC4)) < (phi_f2 * D_809DF5B0)) {
+                this->unk_0144 = 3;
+                this->unk_01B0.y = -3000.0f * D_809DF5B0;
+                if (D_809E0422 != 0) {
+                    this->unk_01B0.y += 3150.0f;
+                }
+                this->unk_0146[0] = 150;
+                this->unk_0164 = 0.0f;
+            }
+            return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_02/func_809DA22C.s")
+        case 3:
+            if (this->unk_0146[0] == 0) {
+                this->unk_0144 = 0;
+            }
+            return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_02/func_809DA24C.s")
+        case 5:
+            this->unk_01B0.x = player->actor.world.pos.x;
+            this->unk_01B0.y = player->actor.world.pos.y + (100.0f * D_809DF5B0);
+            this->unk_01B0.z = player->actor.world.pos.z;
+            if (this->unk_0146[0] == 0) {
+                this->unk_0144 = 3;
+                this->unk_01B0.x = randPlusMinusPoint5Scaled(500.0f * D_809DF5B0) + this->actor.world.pos.x;
+                this->unk_01B0.y = -3000.0f * D_809DF5B0;
+                if (D_809E0422 != 0) {
+                    this->unk_01B0.y += 3150.0f;
+                }
+                this->unk_01B0.z = randPlusMinusPoint5Scaled(500.0f * D_809DF5B0) + this->actor.world.pos.z;
+                this->unk_0146[0] = 150;
+                this->unk_0164 = 0.0f;
+            }
+            return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_02/func_809DA264.s")
+        case 10:
+            if (this->unk_1678 != 0) {
+                this->unk_019A = 0x500;
+                this->unk_01A0 = 4608.0f;
```

Comment:
```suggestion
                this->unk_01A0 = (f32)0x1200;
```
May not need the `(f32)`